spec/manuals_publisher/upload_attachment_spec.rb

```diff
@@ -0,0 +1,55 @@
+feature "Uploading an attachment on Manuals Publisher", manuals_publisher: true do
+  include ManualsHelpers
+
+  let(:title) { title_with_timestamp }
+  let(:section_title) { title_with_timestamp }
+  let(:attachment_title) { title_with_timestamp }
+  let(:file) { File.expand_path("../fixtures/manuals_publisher/manuals.png", __dir__) }
+
+  scenario "Uploading an attachment to a manual" do
+    given_there_is_a_draft_manual_with_a_section
+    when_i_upload_an_attachment
+    and_add_the_attachment_to_the_draft
+    then_i_can_access_the_attachment_through_the_draft
+  end
+
+  def given_there_is_a_draft_manual_with_a_section
+    create_draft_manual(title: title)
+    create_manual_section(title: section_title)
+  end
+
+  def when_i_upload_an_attachment
+    click_link(section_title)
+    click_link("Edit section")
+    click_link("Add attachment")
+
+    fill_in("Title", with: attachment_title)
+    attach_file("File", file)
+    click_button("Save attachment")
+
+    expect(page).to have_content(attachment_title)
+  end
+
+  def and_add_the_attachment_to_the_draft
+    attachment_text = find("span", text: /\[InlineAttachment/).text
+    fill_in("Section body", with: attachment_text)
+
+    click_button("Save as draft")
+  end
+
+  def then_i_can_access_the_attachment_through_the_draft
+    url = find_link("Preview draft")[:href]
+
+    reload_url_until_status_code(url, 200)
```

Comment:
Do you think we're at any risk here of this only being the first draft of the manual and thus not having the edited attachment?

Reply:
Good catch.  We've added an additional check once you've clicked through to the manual contents page, then checking that the updated section version has been published.

Reply:
Thanks 👍